src/dynamics_tendencies.jl

```diff
@@ -0,0 +1,216 @@
+"""
+Compute the spectral tendency of the surface pressure logarithm
+"""
+function surface_pressure_tendency!(Prog::PrognosticVariables{NF}, # Prognostic variables
+                                   Diag::PrognosticVariables{NF}, # Diagnostic variables
+                                   l2::Int,                       # leapfrog index 2 (time step used for tendencies)
+                                   C::Constants{NF}
+                                   ) where {NF<:AbstractFloat}
+
+
+    @unpack pres_surf= Prog 
+    @unpack pres_surf_tend = Diag.Tendencies
+
+    @unpack u_grid,v_grid,div_grid = Diag.gridvars
+    @unpack u_mean,v_mean,d_mean,dumc,px,py = Diag.miscvars
+    @unpack nlev, dhs = C 
+
+    #Initialise mean fields. 
+    u_mean[:,:] = 0.0
```

Comment:
This causes an error in Julia as the broadcasting has to be triggered with the dot-notation `.=` furthermore there's no need to write `0.0` as it will be converted to eltype of `u_mean` anyway. You can write `u_mean[:,:] .= 0`, but the Julian way of doing it is more `fill!(u_mean,0)`, probably doesn't matter much.

Reply:
Corrected to use `fill!`